bolt-modules/boltlib/lib/puppet/functions/run_command.rb

```diff
@@ -56,6 +57,18 @@ def run_command_with_description(command, targets, description = nil, options =
     options = options.transform_keys { |k| k.sub(/^_/, '').to_sym }
     options[:description] = description if description
 
+    # Ensure env_vars is a hash and that each hash value is transformed to JSON
+    # so we don't accidentally pass Ruby-style data to the target.
+    if options[:env_vars]
+      unless options[:env_vars].is_a?(Hash)
+        raise Bolt::ValidationError, "Option 'env_vars' must be a hash"
+      end
+
+      options[:env_vars] = options[:env_vars].transform_values do |val|
+        val.is_a?(Hash) ? val.to_json : val
```

Comment:
Is there a reason not to do this for all types? This becomes weird when, for example, you have an array that includes a hash and that *doesn't* get converted:
```
lucy ~/githubs/bolt ((40f60cdb...)) $ be bolt plan run bolt -t localhost
...
Finished on localhost:
  STDOUT:
    ["hi", "there", {"hi"=>"there"}]
```

Reply:
I think we actually do want to do it for arrays, too. My hesitance with doing it for all types is that it changes existing behavior for string values. Converting strings to JSON will add quotes to the beginning and end of the string, which might affect users using this option...

Reply:
Sorry I completely missed your comment above! Agreed.